src/strategies/erc721-with-method-weight/index.ts

```diff
@@ -0,0 +1,46 @@
+import { multicall } from '../../utils';
+import { Options } from './types';
+
+export const author = 'pro100skm';
+export const version = '0.0.1';
+
+const abi = (methodName: string) => [
+  `function ${methodName}(address addr, uint256 level) public view returns (uint256)`
+];
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options: Options,
+  snapshot
+) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+  const { address: contractAddress, methodName, levelsWeight } = options;
+
+  const keys = Object.keys(levelsWeight);
+
+  let weightSum = Array(keys.length).fill(0);
+
+  for (let i = 0; i < keys.length; i++) {
```

Comment:
Need to add a limit to these `keys` else it may cause some memory issues, and throw an error if it receives more keys

Reply:
I just simplified the logic of fetching weights